compiler/noirc_evaluator/src/acir/tests/intrinsics.rs

```diff
@@ -44,6 +44,56 @@ fn slice_push_back() {
     ");
 }
 
+#[test]
+fn slice_push_back_unknown_length() {
+    // Here we use v2 as the length of the slice to show the generated ACIR when the
+    // length is now known at compile time.
+    let src = "
+    acir(inline) predicate_pure fn main f0 {
+      b0(v0: u32, v1: u32, v2: u32):
+        v3 = make_array [Field 2, Field 3] : [Field]
+        // Mutate it at index v0 (to make it non-constant in the circuit)
+        v5 = array_set v3, index v0, value Field 4
+        v8, v9 = call slice_push_back(v2, v5, Field 10) -> (u32, [Field])
+        constrain v8 == v1
+        // Mutate the new slice to make the result block observable
+        v11 = array_set v9, index v0, value Field 20
+        return
+    }
+    ";
+    let program = ssa_to_acir_program(src);
+
+    assert_circuit_snapshot!(program, @r"
+    func 0
+    private parameters: [w0, w1, w2]
+    public parameters: []
+    return values: []
+    BLACKBOX::RANGE input: w1, bits: 32
+    ASSERT w3 = 2
+    ASSERT w4 = 3
+    INIT b1 = [w3, w4]
+    ASSERT w5 = 4
+    WRITE b1[w0] = w5
+    ASSERT w6 = 0
+    READ w7 = b1[w6]
+    ASSERT w8 = 1
+    READ w9 = b1[w8]
+    INIT b2 = [w7, w9, w6]
+    ASSERT w10 = 10
+    WRITE b2[w2] = w10
+    ASSERT w2 = w1 - 1
+    READ w11 = b2[w6]
+    READ w12 = b2[w8]
+    READ w13 = b2[w3]
+    READ w14 = b2[w6]
+    READ w15 = b2[w8]
+    READ w16 = b2[w3]
```

Comment:
@guipublic I added a test to see the generated ACIR in the new case.

As you can see, it seems we read from `b2` twice (here we read at positions 0, 1 and 2, twice).
(this might be easier to notice with https://github.com/noir-lang/noir/pull/10068 but that fails for a reason I don't understand)

Just mentioning because there might be an opportunity to optimize this, reducing the memory performance loss (though I'm not sure if it'll affect it in practice).

Reply:
Aaaah... it's because `slice_push_back` creates a new block ID without any relation to any SSA value, but it could be related to the result.

I'll push a commit to try this, if it doesn't work we can revert it.